routers/admin/auths.go

```diff
@@ -177,7 +177,7 @@ func parseSSPIConfig(ctx *context.Context, form auth.AuthenticationForm) (*model
 		return nil, errors.New(ctx.Tr("form.SSPISeparatorReplacement") + ctx.Tr("form.alpha_dash_dot_error"))
 	}
 
-	if !langCodePattern.MatchString(form.SSPIDefaultLanguage) {
+	if !util.IsEmptyString(form.SSPIDefaultLanguage) && !langCodePattern.MatchString(form.SSPIDefaultLanguage) {
```

Comment:
If you use `IsEmptyString()` here, you must use `TrimSpace()` below, as the first counts spaces as empty. Otherwise `DefaultLanguage = "    "` is possible.

Reply:
Fixed with https://github.com/go-gitea/gitea/pull/8463/commits/814873aace96c9e62af7dbb34309b76a08be9b85